cartographer_ros/cartographer_ros/node.cc

```diff
@@ -45,22 +47,25 @@ namespace carto = ::cartographer;
 
 using carto::transform::Rigid3d;
 
+namespace {
+constexpr int kInfiniteSubscriberQueueSize = 0;
+constexpr double kTfBufferCacheTimeInSeconds = 1e6;
 constexpr int kLatestOnlyPublisherQueueSize = 1;
 
-Node::Node(const MapOptions& map_options, tf2_ros::Buffer* const tf_buffer)
-    : map_options_(map_options), map_builder_bridge_(map_options_, tf_buffer) {}
+TrajectoryOptions LoadTrajectoryOptions(const string& code) {
+  auto file_resolver = cartographer::common::make_unique<
+      cartographer::common::ConfigurationFileResolver>(
+      std::vector<string>{""});
+  cartographer::common::LuaParameterDictionary lua_parameter_dictionary(
```

Comment:
After understanding the code a bit better, I think this should not be needed if `TrajectoryOptions` is converted to a ROS message with a binary proto.